### ISSUE TITLE
owner and moderator info lost on upgrade to 6.2.20 or 6.2.22

### ISSUE THREAD

**Issue by pkissman** [sympa-community/sympa#71]
Having upgraded to either 6.2.20 or 6.2.22, neither sympa nor wwsympa can find existing list owners and moderators.  One has to edit the list ownership info and save it again (edit_list_request) in order for list ownership to be found. 

Log file example:

notice Sympa::Spindle::ToModeration::_send_confirm_to_editor() **Warning: No owner and editor defined at all in list** Sympa::List <mylist@mydomain>
err main::#243 > Sympa::Spindle::spin#92 > Sympa::Spindle::ToModeration::_twist#52 > Sympa::Spindle::ToModeration::_send_confirm_to_editor#147 **Impossible to send the moderation request for message Sympa::Message**
<mylist@mydomain.1506099454.6074> to editors of list Sympa::List <mylist@mydomain> Neither editor nor owner defined!
Sep 22 12:57:35 mblc sympa_msg[5113]: err main::#243 > Sympa::Spindle::spin#92 > Sympa::Spindle::ToModeration::_twist#55 Failed to send moderation request of Sympa::Message <mylist@mydomain.1506099454.6074> from paul.kissman@state.ma.us for list Sympa::List <mylist@mydomain>to editor(s)
****************

**Comment by pkissman** [comment]
I just realized that my whole admin_table had been wiped out by the sympa upgrade process, which is why no owners or moderators were found.

I restored from a backup and things look good.

This seems to happen due to the upgrade script:

## Empty the admin_table entries and recreate them
    $log->syslog('notice', 'Rebuilding the admin_table...');
    Sympa::List::delete_all_list_admin();
    foreach my $list (@$all_lists) {
        $list->sync_include_admin();

Not sure why the list doesn't get restored.  My admins are not in a separate included source.

**Comment by racke** [comment]
Really, we shouldn't delete a whole table without safety belt. At the very least, a dump should be stored in case it goes wrong.

Unclear why the rebuild doesn't work. @pkissman, what was the previous Sympa versions and which database system and version do you use?

Also can you check your log for messages related to the admin_table rebuild event?

**Comment by pkissman** [comment]
Stefan,

I first encountered this problem when I upgraded from 6.2.18 to 6.2.20, and then again when going from 6.2.20 to 6.2.22 today.

Using mysql (5.5.57-0+deb7u1)

I get the standard notice in my syslog or messages log:

Oct  2 10:52:01 mlinwww02 sympa[9139]: notice Sympa::Upgrade::upgrade() Rebuilding the admin_table...

Unfortunately, I can’t seem to find anything else (mysql, system logging, or sympa logs)

Pjk

**Comment by ikedas** [comment]
@pkissman , though i don't know why, list config files might be broken.  Restore them by doing;
1. Stop services, if possible.
2. Go to list directory, and overwrite `config` file of each list with the recent version of unbroken backup files (named config._number_ and placed in the same directory as `config`).
3. Remove `config.bin`, if any.
4. Start services, if you have stopped.

Then see if owners/moderators are restored in configuration page.

**Comment by pkissman** [comment]
Thank you for the solution you’ve suggested.

Unfortunately, I think I will have to wait for the next upgrade to give this a try.

I had already restored the admin_table from a backup, which solved the problem until the next upgrade.

I did try to follow your advice.  I deleted my config.bin files and took older version of the config files for each list (e.g  config.20) and copied to config

I reinstalled sympa over the old installation and restarted.

But since  there was no version change, (6.2.22 => 6.2.22) the upgrade wizard didn’t feel the need to do anything.  So the admin_table was not cleared.  And the problem did not reappear.

pjk

**Comment by racke** [comment]
OK, I can confirm this problem. I'm working on a Sympa migration and it blew away my admin table as well. Too bad really 😠

**Comment by racke** [comment]
```
notice Sympa::Upgrade::upgrade() Rebuilding config.bin files for ALL lists...it may take a while...
notice Sympa::Upgrade::upgrade() Rebuilding the admin_table...
notice Sympa::Upgrade::upgrade() Number of lists: 0
```

Oh well, so it is just deleting and not adding it back (last log message added locally).
Will investigate further.

**Comment by racke** [comment]
My analysis:

` my $all_lists = Sympa::List::get_lists('*', 'reload_config' => 1);`

With that call, Sympa doesn't consult the database but goes into the list_data directories of all the robots and searches for files named "config". 

If it doesn't find these files, Sympa doesn't put back in the admins into the admin table.

Do you use robots in your setup, @pkissman ?

**Comment by ikedas** [comment]
> If it doesn't find these files, Sympa doesn't put back in the admins into the admin table.

I see, but why get_lists() omits existing lists? @racke, can you think of anything?

**Comment by racke** [comment]
Hard to tell, insufficient permissions maybe.

**Comment by ikedas** [comment]
Ok, I'll merge your PR. Let's wait for the next chance of upgrading.

**Comment by pkissman** [comment]
Stefan,


I just run lists on a single domain. There are no virtual hosts or robot.conf files.  Thanks for asking.  I'm not sure how that plays into your analysis below.

I tend to do upgrades as a superuser and then change ownership of files back to the sympa user after the install.  so anything in the upgrade script should have sufficient permissions to read and write wherever it needs to.

pjk



________________________________

**Comment by ikedas** [comment]
I encountered this bug upgrading 6.2.22 to 6.2.23b.1.  Reloading httpd and sympa services seemed to restore owners/editors.
A commit above may avoid it.

**Comment by ikedas** [comment]
In my side, the bug was not be reproduced upgrading 6.2.23b.1 to 6.2.23b.2.

**Comment by ikedas** [comment]
Fix to issue #11 will make worry on synchronization between config and database (see also issue #49).  When it is closed, this issue will be closed.

**Comment by ikedas** [comment]
After a fix on #11, owners/moderators will no longer removed from database during upgrade.  Therefore this issue is closed.
